app/javascript/article-form/components/imageUploadHelpers.js

```diff
@@ -0,0 +1,82 @@
+import { handleImageFailure } from './dragAndDropHelpers';
+
+// Placeholder text displayed while an image is uploading
+const UPLOADING_IMAGE_PLACEHOLDER = '![Uploading image](...)';
+
+export function handleImageUploading(textAreaRef) {
```

Comment:
**(non-blocking, suggestion):** Perhaps adding some method documentation to the functions will be helpful.